plugins/UsersManager/Controller.php

```diff
@@ -387,37 +387,67 @@ public function addNewToken()
             throw new Exception('Not allowed');
         }
 
-        $noDescription = false;
+        $invalidExpireDate = null;
+        if (!empty($_POST['token_expire_date'])) {
+            $tokenExpireDate = \Piwik\Request::fromRequest()->getStringParameter('token_expire_date');
+
+            try {
+                $invalidExpireDate = true;
+                if (preg_match('/^(\d{4})-(\d{2})-(\d{2})$/', $tokenExpireDate)) {
+                    Date::factory($tokenExpireDate);
```

Comment:
Maybe some check if the date is actually in the future might be helpful. Otherwise someone might by accident create a token that is already expired and would never show up in the list of tokens afterwards.